aws/amazon-sagemaker-examples
# Patch for debugger notebooks with profiling

*Issue #, if available:*

*Description of changes:*
Various necessary bug fixes in the notebooks.
- Upgrade to the latest sagemaker SDK in all of the notebooks
- Install smdebug in all notebooks where it used
- Update the profiler config for TF 2.3.1 jobs to specify the local path for framework profiling
- Use the image URI for the DLC image directly instead of relying on framework and python version in the estimator. Dynamically determine the region to complete the image URI.
- Swap the values for `instance_type` and `instance_count` in the bottleneck notebooks.
- Remove the array wrapped around the rules specified in the estimator.
- Initialize the system metrics reader wherever it is used.
- Lower the batch size in the training job for the callback notebook.

After applying the above fixes, I tested in CMH (us-east-2). All of the notebooks execute correctly. All of the training jobs from the notebooks pass except for the dataset bottleneck notebook, which fails due to an obscure error. For now, we will exclude this notebook and add it back again once we have fixed the notebook.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
